### PR TITLE
Refactor command result

### PR DESCRIPTION
**Pull request by chrisjwelly**
Adds subclasses that corresponds to the CommandResults of the different commands

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/edit/EditInternshipCommand.java [hunk 1]
```diff
@@ -79,7 +80,11 @@ public CommandResult execute(Model model) throws CommandException {
         } catch (DuplicateItemException e) {
             throw new CommandException(MESSAGE_DUPLICATE_ITEM);
         }
-        return new CommandResult(editedInternship.toString(),
+
+        model.setInternship(toEdit, editedInternship);
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/edit/EditInternshipCommand.java]: Extra set

File: src/main/java/seedu/address/logic/commands/edit/EditProjectCommand.java [hunk 2]
```diff
@@ -75,7 +76,12 @@ public CommandResult execute(Model model) throws CommandException {
         } catch (DuplicateItemException e) {
             throw new CommandException(MESSAGE_DUPLICATE_ITEM);
         }
-        return new CommandResult(editedProject.toString(), String.format(MESSAGE_EDIT_PROJECT_SUCCESS, editedProject),
+
+        model.setProject(toEdit, editedProject);
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/edit/EditProjectCommand.java]: Extra set

File: src/main/java/seedu/address/logic/commands/edit/EditResumeCommand.java [hunk 3]
```diff
@@ -62,7 +63,12 @@ public CommandResult execute(Model model) throws CommandException {
         } catch (DuplicateItemException e) {
             throw new CommandException(MESSAGE_DUPLICATE_ITEM);
         }
-        return new CommandResult(editedResume.toString(), String.format(MESSAGE_EDIT_RESUME_SUCCESS, editedResume),
+
+        model.setResume(toEdit, editedResume);
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/edit/EditResumeCommand.java]: Extra set

File: src/main/java/seedu/address/logic/commands/edit/EditSkillCommand.java [hunk 4]
```diff
@@ -66,7 +67,13 @@ public CommandResult execute(Model model) throws CommandException {
         } catch (DuplicateItemException e) {
             throw new CommandException(MESSAGE_DUPLICATE_ITEM);
         }
-        return new CommandResult(editedSkill.toString(), String.format(MESSAGE_EDIT_SKILL_SUCCESS, editedSkill),
+
+        model.setSkill(toEdit, editedSkill);
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/edit/EditSkillCommand.java]: Extra set

File: src/main/java/seedu/address/logic/commands/results/CommandResult.java [hunk 5]
```diff
@@ -8,52 +8,33 @@
...
-
-    // TODO: OOP to make more CommandResult classes
-
-    private final String feedbackToUser;
-    private final String dataToUser;
-    private final String displayType;
-
     /** Preview information about a resume. */
-    private final boolean isShowPreview;
+    protected boolean isShowPreview;
```
Comment (helloImHai) [inline src/main/java/seedu/address/logic/commands/results/CommandResult.java]: Should we set these values with initial value of false? I'm scared of a null returning when it's supposed to be a boolean. But actually iirc, if (null) is false.. so it's probably okay.

**Review by helloImHai — CHANGES_REQUESTED** [review]
As it should be m8 💯💯💯
Just remove the extra set values in EditXXXCommand

File: src/main/java/seedu/address/logic/commands/results/CommandResult.java (hunk 5 above)
Comment (chrisjwelly) [reply src/main/java/seedu/address/logic/commands/results/CommandResult.java]: Good idea. Working on it!

**Review by helloImHai — APPROVED** [review]
All fine and dandy 💃💃
Blame Duong for the merge conflict lmao

**Pull request merged**